instance/models/load_balancer.py

```diff
@@ -210,22 +264,28 @@ def reconfigure(self, triggering_instance_id=None):
         reconfiguration of the load balancer.
         """
         self.logger.info("Reconfiguring load-balancing server %s", self.domain)
-        with self._configuration_lock():
-            self.run_playbook(self.get_ansible_vars(triggering_instance_id))
+        self.run_playbook(self.get_ansible_vars(triggering_instance_id))
 
+    @modifies_configuration
     def deconfigure(self):
         """
         Remove the configuration fragment from the load-balancing server.
         """
+        self.logger.info("Deconfiguring load-balancing server %s", self.domain)
         fragment_name = settings.LOAD_BALANCER_FRAGMENT_NAME_PREFIX + self.fragment_name_postfix
-        with self._configuration_lock():
-            self.run_playbook(
-                "FRAGMENT_NAME: {fragment_name}\nREMOVE_FRAGMENT: True".format(fragment_name=fragment_name)
-            )
+        self.run_playbook(
+            "FRAGMENT_NAME: {fragment_name}\nREMOVE_FRAGMENT: True".format(fragment_name=fragment_name)
+        )
```

Comment:
I didn't realize this in the first version, but adding the same decorator here doesn't really do what we want.  We want to remove the fragment, not mark the LB for reconfiguration.  If we want to remove the fragment, we should just block until we are allowed to, so basically do exactly what we did before.

Reply:
Ah, OK!